core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -331,6 +331,30 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * A file name may contain some invalid url characters, such as " ". This method will convert the
+   * file name to a raw path accepted by `java.net.URI(String)`.
+   *
+   * Note: the file name must not contain "/" or "\"
+   */
+  def encodeFileNameToURIRawPath(fileName: String): String = {
```

Comment:
I don't think that file name is a problem in a URL though. `:` is not reserved in that part. You'd have to send this to `java.net.URI` "manually" like you've done, or else use `file:/abc:xyz`. Then it's doing the same thing; your method wouldn't escape the colon either.